beta-src/src/utils/map/drawArrow.ts

```diff
@@ -9,105 +10,106 @@ export default function drawArrow(
    */
   arrowIdentifier: string,
   arrowType: ArrowType,
-  receiverTerritory: Territory,
+  arrowColor: ArrowColor,
+  receiverType: "territory" | "unit" | "arrow",
+  receiverIdentifier: Territory | string,
   unitTerritory: Territory,
 ): void {
   const d3MapSelector = d3.select("#map");
 
-  const fromTerritoryName = Territory[unitTerritory];
-  const toTerritoryName = Territory[receiverTerritory];
-
-  let unitSlotEl = d3.select(`#${fromTerritoryName}-unit`).node();
-
-  const toTerritoryReceiver: SVGRectElement = d3
-    .select(`#${toTerritoryName}-arrow-receiver`)
-    .node();
-
-  const fromTerritoryEl: SVGSVGElement = d3
-    .select(`#${fromTerritoryName}-territory`)
-    .node();
-
-  const toTerritoryEl: SVGSVGElement = d3
-    .select(`#${toTerritoryName}-territory`)
-    .node();
-
-  if (fromTerritoryEl && toTerritoryEl && toTerritoryReceiver && unitSlotEl) {
-    unitSlotEl = unitSlotEl.parentNode;
-    const unitSlotElBBox = unitSlotEl.getBBox();
-    const unitSlotElX = Number(unitSlotEl.getAttribute("x")) + unitSlotElBBox.x;
-
-    let x1 = Number(fromTerritoryEl.getAttribute("x")) + unitSlotElX;
-
-    const x2 =
-      Number(toTerritoryEl.getAttribute("x")) +
-      Number(toTerritoryReceiver.getAttribute("x"));
-
-    let y1 =
-      Number(fromTerritoryEl.getAttribute("y")) +
-      Number(unitSlotEl.getAttribute("y")) +
-      unitSlotElBBox.y;
-
-    const y2 =
-      Number(toTerritoryEl.getAttribute("y")) +
-      Number(toTerritoryReceiver.getAttribute("y"));
-
-    const w = unitSlotElBBox.width;
-    const h = unitSlotElBBox.height;
-
-    const xDiff = x2 - x1;
-    const yDiff = y2 - y1;
-
-    const positionChangeBuffer = 75;
-
-    if (Math.abs(xDiff) < positionChangeBuffer && yDiff < 0) {
-      // top center
-      x1 += w / 2;
-    } else if (Math.abs(xDiff) < positionChangeBuffer && yDiff > 0) {
-      // bottom center
-      y1 += h;
-      x1 += w / 2;
-    } else if (
-      xDiff > positionChangeBuffer &&
-      yDiff < 0 &&
-      Math.abs(yDiff) > positionChangeBuffer
-    ) {
-      // top right
-      x1 += w;
-    } else if (
-      xDiff > positionChangeBuffer &&
-      Math.abs(yDiff) < positionChangeBuffer
-    ) {
-      // center right
-      y1 += h / 2;
-      x1 += w;
-    } else if (xDiff > positionChangeBuffer && yDiff > positionChangeBuffer) {
-      // bottom right
-      y1 += h;
-      x1 += w;
-    } else if (xDiff < 0 && Math.abs(yDiff) < positionChangeBuffer) {
-      // center left
-      y1 += h / 2;
-    } else if (xDiff < 0 && yDiff > positionChangeBuffer) {
-      // bottom left
-      y1 += h;
+  switch (receiverType) {
```

Comment:
This is in a switch case but we only have the default case currently. Are we going to be adding more cases?